nucypher/characters/lawful.py

```diff
@@ -163,6 +165,31 @@ def get_policy_pubkey_from_label(self, label: bytes) -> UmbralPublicKey:
         policy_pubkey = alice_delegating_power.get_pubkey_from_label(label)
         return policy_pubkey
 
+    def revoke(self, policy):
+        """
+        Parses the treasure map and revokes arrangements in it.
+        If any arrangements can't be revoked, then the node_id and
+        arrangement_id are added to a list and returned. If a node returns a
+        404 status code, then it will add a tuple of the Revocation and `None`.
+        """
+        try:
+            # Wait for a revocation threshold of nodes to be known ((n - m) + 1)
+            revocation_threshold = ((policy.n - policy.treasure_map.m) + 1)
+            self.block_until_specific_nodes_are_known(
+                    policy.revocation_kit.revokable_addresses,
+                    allow_missing=revocation_threshold)
+        except self.NotEnoughTeachers as e:
+            raise e
+        else:
+            failed_revocations = dict()
+            for node_id in policy.revocation_kit.revokable_addresses:
+                ursula = self.known_nodes[node_id]
+                revocation = policy.revocation_kit[node_id]
+                response = self.network_middleware.revoke_arrangement(ursula, revocation)
+                if response.status_code != 200:
+                    failed_revocations[node_id] = (revocation, response.status_code)
+        return failed_revocations
```

Comment:
what if `len(failed_revocations) < n-m+1`? Shouldn't we raise an exception in that case (e.g., `NotEnoughRevocations`)?

Reply:
Yes, I think _that_ is the right approach

Reply:
https://github.com/nucypher/nucypher/issues/595